### ISSUE TITLE
Trying to access Realm Object from an interface

### ISSUE THREAD

**Issue by moderateepheezy** [realm/realm-java#3363]
am using Realm, and i created an interface to parse RealmObject in my Adapter to be implemented in an activity but i get this error

**Object is no longer valid to operate on. Was it deleted by another thread?**

**NewsAdapter.java**

```
`
final NewsTrend postsData = getItem(position);
 holder.ivLike.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                eventListener.onLikeClick(v, postsData, holder.tvNewsCountLike, holder.ivLike);
            }
        });
`
```

Implementing the Listener in Fragment

**Fragment.java**
```
`
@Override
    public void onLikeClick(View view, NewsTrend postData, TextView tvNewsCountLike,
                            ImageView ivLike) {
        if (!postData.isChecked()) {
            realm.beginTransaction();
            postData.setChecked(true);
            realm.commitTransaction();
            tvNewsCountLike.setText("" + (Integer.parseInt(tvNewsCountLike.getText().toString()) + 1));
            ivLike.setImageResource(R.drawable.kalp_dolu_kucuk);
        } else {
            realm.beginTransaction();
            postData.setChecked(false);
            realm.commitTransaction();
            tvNewsCountLike.setText("" + (Integer.parseInt(tvNewsCountLike.getText().toString()) - 1));
            ivLike.setImageResource(R.drawable.kalp_bos_kucuk);
        }
        adapter.notifyDataSetChanged();
        like(postData.getNews_id());
    }
```

**Comment by Zhuinden** [comment]
What version of Realm are you using?
Because you're using it as if it were Realm 0.82.2
Even though the latest version is 1.2.0
And you're kinda neglecting changes that were done in 0.89.0

By which I mean you're writing synchronously on the UI thread and immediately calling `adapter.notifyDataSetChanged()` afterwards, which is bound to fail; you're supposed to keep a field reference to a `RealmChangeListener` to the table you're notifying (`RealmResults<PostData>`) which calls `adapter.notifyDataSetChanged()`.

Then again, this code doesn't show anything related to deleting data or closing Realms, so your problem lies elsewhere.

**Comment by Zhuinden** [comment]
This is based on the following stack overflow question 

http://stackoverflow.com/questions/39211133/trying-to-access-realm-object-from-an-interface?noredirect=1#comment65764295_39211133

Realm version is 0.85.0

**Comment by moderateepheezy** [comment]
@Zhuinden i posted in on stack overflow, i have been on this issue for almost a week, and i haven't been able to solve,
so please i really need your help, how will i approach the solution to the problem i am having, cus its taking too long.

I don't mind sharing my full code for better understanding of what am trying to do.

**Comment by beeender** [comment]
@moderateepheezy It looks like the issue we fixed in #2990 . Let's keep the conversation on SO. And I am closing this issue now.

**Comment by Zhuinden** [comment]
@beeender in that case this issue was fixed in 1.0.1
